tests/structured-data/class-archive-post-test.php

```diff
@@ -233,6 +233,143 @@ public function test_term_archive() {
+		// See https://github.com/Parsely/wp-parsely/issues/151
+		$this->set_permalink_structure( '/%postname%/' );
+
+		// Setup Parsley object.
+		$parsely         = new \Parsely();
+		$parsely_options = get_option( \Parsely::OPTIONS_KEY );
+
+		// Register Post Type with specific archive URL.
+		register_post_type(
+			'custom_post_type',
+			array(
+				'public'      => true,
+				'has_archive' => 'cpt-tax-archive',
+			) 
+		);
+
+		// Add post to custom post type.
+		self::factory()->post->create(
+			array(
+				'title'     => 'Post Title',
+				'post_type' => 'custom_post_type',
+			)
+		);
+
+		// Flush rewrite rules after creating new post type with archive.
+		flush_rewrite_rules();
+
+		// Go to the custom post type archive page.
+		$this->go_to( home_url( '/cpt-tax-archive' ) );
+
+		// The query should be for a custom post type archive.
+		self::assertQueryTrue( 'is_archive', 'is_post_type_archive' );
+
+		// Create the structured data for that category.
+		// The category metadata doesn't use the post data, but the construction method requires it for now.
+		$structured_data = $parsely->construct_parsely_metadata( $parsely_options, get_post() );
+
+		// Check the required properties exist.
+		$this->assert_data_has_required_properties( $structured_data );
+
+		// The headline should be the category name.
+		self::assertEquals( 'custom_post_type', $structured_data['headline'] );
+		self::assertEquals( home_url( '/cpt-tax-archive' ), $structured_data['url'] );
+	}
+
+	/**
+	 * Check metadata for custom post type term archive.
+	 *
+	 * @covers \Parsely::construct_parsely_metadata
+	 * @uses \Parsely::__construct
+	 * @uses \Parsely::get_author_name
+	 * @uses \Parsely::get_author_names
+	 * @uses \Parsely::get_bottom_level_term
+	 * @uses \Parsely::get_category_name
+	 * @uses \Parsely::get_clean_parsely_page_value
+	 * @uses \Parsely::get_coauthor_names
+	 * @uses \Parsely::get_current_url
+	 * @uses \Parsely::get_first_image
+	 * @uses \Parsely::get_options
+	 * @uses \Parsely::get_tags
+	 * @uses \Parsely::post_has_trackable_status
+	 * @uses \Parsely::update_metadata_endpoint
+	 * @group metadata
+	 */
+	public function test_term_archive_custom_post_type() {
+		// Set permalinks, as Parsely currently strips ?page_id=... from the URL property.
+		// See https://github.com/Parsely/wp-parsely/issues/151
+		$this->set_permalink_structure( '/%postname%/' );
+
+		// Setup Parsley object.
+		$parsely         = new \Parsely();
+		$parsely_options = get_option( \Parsely::OPTIONS_KEY );
+
+		// Register custom post type and custom taxonomy for that post type.
+		register_post_type( 'custom_post_type', array( 'public' => true ) );
+		register_taxonomy( 'custom_tax', array( 'custom_post_type' ) );
+
+		// Insert a single term, and a Post in that custom post type with the custom term.
+		$term    = self::factory()->term->create(
+			array(
+				'taxonomy' => 'custom_tax',
+				'slug'     => 'term',
+				'name'     => 'Custom Taxonomy',
```

Comment:
```suggestion
				'name'     => 'Custom Taxonomy Term',
```

It's easy to get confused between the name of the taxonomy, and the name of a term in that taxonomy, so keep the distinction clear.

The assertion on this name will also need updating.